src/js/AccessibilityUtils.js

```diff
@@ -1051,22 +1051,25 @@ axs.utils.getRoles = function(element, implicit) {
     if (!roleValue)  // role='' or implicit role came up empty
         return null;
     var roleNames = roleValue.split(' ');
-    var roles = [];
-    var valid = true;
+    var result = { roles: [], valid: false };
     for (var i = 0; i < roleNames.length; i++) {
         var role = roleNames[i];
         var ariaRole = axs.constants.ARIA_ROLES[role];
-        if (ariaRole && !ariaRole.abstract) {
-            var roleObject = {'name': role, 'details': axs.constants.ARIA_ROLES[role], 'valid': true};
-            roles.push(roleObject);
+        var roleObject = {'name': role};
+        if (ariaRole && !ariaRole['abstract']) {
```

Comment:
In days of yore `abstract` was a "future reserved word" and the bracket notation would "hide" it since it is just a string. I actually don't remember why I changed it, I must have been having a flashback. I'll put it back how it was since [it is no longer reserved](http://www.ecma-international.org/ecma-262/5.1/#sec-7.6.1.2).